test/integration/builder/transaction.js

```diff
@@ -659,5 +659,34 @@ module.exports = function(knex) {
         return builder;
       });
     });
+
+    it('#3690 does not swallow exception when error during transaction occurs', async () => {
+      const killConnectionMap = {
+        mysql: async (connection) => knex.raw('KILL ?', [connection.threadId]),
+        mysql2: async (connection) => knex.raw('KILL ?', [connection.threadId]),
+        pg: async (connection) =>
+          knex.raw('SELECT pg_terminate_backend(?)', [connection.processID]),
+        redshift: async (connection) =>
+          knex.raw('SELECT pg_terminate_backend(?)', [connection.processID]),
+        mssql: async (connection, trx) => {
+          const id = (await trx.raw('select @@SPID as id'))[0].id;
+          return knex.raw(`KILL ${id}`);
+        },
+        oracledb: async (connection, trx) => {
+          const id = (await trx.raw(
+            `select sys_context('userenv','sessionid') id from dual;`
+          ))[0].id;
+          return knex.raw(`Alter System Kill Session '${id}'`);
+        },
+      };
```

Comment:
So: it looks like this logic is already encapsulated in a `.cancelQuery(..)` method for the various clients that support it.  Ex:

https://github.com/knex/knex/blob/b025aea31881e9a08c67ae0ccb6a4c1ceeae2ef2/lib/dialects/mysql/index.js#L172-L184

So, we can probably just call that method instead.

Additionally, Clients that support this functionality also set `canCancelQuery = true`.  So, this means we can probably also do:

```
if(!knex.client.canCancelQuery) {
  return this.skip();
}
```

Reply:
This is not cancelling query, this is killing connection/session

Reply:
Ah -- good call.  I just saw the `KILL` keyword and jumped to conclusions.

Anyhoo: for now, maybe we should limit the scope to the handful of clients that we've been able to test successfully  (mysql, mysql2, pg, and maybe redshift).  We can then mark the others as "pending" and follow up on this after the `0.20.11` release.

@maximelkin + @kibertoad : Thoughts?

Reply:
+1
fix mssql and oracle unhandled rejections may require more hours to fix

Reply:
Sounds good :)